plugin/evm/limitorders/build_block_pipeline.go

```diff
@@ -149,10 +149,29 @@ func (pipeline *BuildBlockPipeline) runLiquidations(liquidablePositions []Liquid
 			}
 			// @todo: add a restriction on the price range that liquidation will occur on.
 			// An illiquid market can be very adverse for trader being liquidated.
+			fulfillPrice := oppositeOrder.Price
+			spreadLimit := pipeline.configService.getLiquidationSpreadThreshold(liquidable.Market)
+			upperbound := divideByBasePrecision(new(big.Int).Mul(underlyingPrices[liquidable.Market], new(big.Int).Add(_1e6, spreadLimit)))
+			lowerbound := big.NewInt(0)
+			if spreadLimit.Cmp(_1e6) == -1 {
+				lowerbound = divideByBasePrecision(new(big.Int).Mul(underlyingPrices[liquidable.Market], new(big.Int).Sub(_1e6, spreadLimit)))
+			}
+
+			if fulfillPrice.Cmp(upperbound) == 1 || fulfillPrice.Cmp(lowerbound) == -1 {
+				// log.Error("liquidation price out of bound", "fulfillPrice", fulfillPrice, "upperbound", upperbound, "lowerbound", lowerbound)
+				continue
+			}
+
 			fillAmount := utils.BigIntMinAbs(liquidable.GetUnfilledSize(), oppositeOrder.GetUnFilledBaseAssetQuantity())
 			if fillAmount.Sign() == 0 {
 				continue
 			}
+			// while setting liquidation threshold of a position, we do not ensure whether it is a multiple of minSize.
+			// we will take care of that here
+			minSize := pipeline.configService.getMinSizeRequirement(liquidable.Market)
+			fillAmount.Div(fillAmount, minSize)
```

Comment:
should we do it while setting liquidation threshold instead?

https://github.com/hubble-exchange/hubblenet/blob/9ea6cdf7fd24bdcc168c95075713176c50c86c3b/plugin/evm/limitorders/memory_database.go#L687

Reply:
looks good now